src/Umbraco.Web/Routing/DomainUtilities.cs

```diff
@@ -150,15 +150,10 @@ internal static DomainAndUri SelectDomain(IEnumerable<Domain> domains, Uri uri,
         {
             // sanitize the list to have proper uris for comparison (scheme, path end with /)
             // we need to end with / because example.com/foo cannot match example.com/foobar
-            // we need to order so example.com/foo matches before example.com/
-            var domainsAndUris = domains
-                .Where(d => d.IsWildcard == false)
-                .Select(d => new DomainAndUri(d, uri))
-                .OrderByDescending(d => d.Uri.ToString())
```

Comment:
Ordering by URL is only required for `SelectByBase()`, so more specific URLs get matched first and therefore removed here to keep the correct domain sort order.